PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/runner/webapp/AzureDotNetWebAppMvpModel.kt

```diff
@@ -0,0 +1,105 @@
+package com.microsoft.intellij.runner.webapp
+
+import com.microsoft.azure.management.Azure
+import com.microsoft.azure.management.appservice.OperatingSystem
+import com.microsoft.azure.management.appservice.PricingTier
+import com.microsoft.azure.management.appservice.WebApp
+import com.microsoft.azuretools.authmanage.AuthMethodManager
+import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel
+import com.microsoft.azuretools.core.mvp.ui.base.SchedulerProviderFactory
+import rx.Observable
+
+object AzureDotNetWebAppMvpModel {
+
+    //region Web App
+
+    fun createWebAppWithNewAppServicePlan(subscriptionId: String,
+                                          webAppName: String,
+                                          appServicePlanName: String,
+                                          pricingTier: PricingTier,
+                                          region: String,
+                                          isCreatingResourceGroup: Boolean,
+                                          resourceGroupName: String): WebApp {
+        val azure = AuthMethodManager.getInstance().getAzureClient(subscriptionId)
+        return webAppWithNewAppServicePlan(azure, webAppName, appServicePlanName, pricingTier, region, isCreatingResourceGroup, resourceGroupName).create()
+    }
+
+    fun createWebAppWithExistingAppServicePlan(subscriptionId: String,
+                                               webAppName: String,
+                                               appServicePlanId: String,
+                                               isCreatingResourceGroup: Boolean,
+                                               resourceGroupName: String): WebApp {
+
+        val azure = AuthMethodManager.getInstance().getAzureClient(subscriptionId)
+        return webAppWithExistingAppServicePlan(azure, webAppName, appServicePlanId, isCreatingResourceGroup, resourceGroupName).create()
+    }
+
+    fun getDotNetWebApp(subscriptionId: String, webAppId: String): WebApp {
+        return AzureWebAppMvpModel.getInstance().getWebAppById(subscriptionId, webAppId)
+    }
+
+    //endregion Web App
+
+    //region Check Existence
+
+    fun checkResourceGroupExistence(subscriptionId: String, resourceGroupName: String): Boolean {
+        val azure = AuthMethodManager.getInstance().getAzureClient(subscriptionId)
+        return azure.resourceGroups().checkExistence(resourceGroupName)
+    }
+
+    fun checkResourceGroupExistenceAsync(subscriptionId: String, resourceGroupName: String): Observable<Boolean> {
```

Comment:
unused, let's remove

Reply:
Added a comment about the method. Add @Suppress("unused") for now because it might be helpful in future in other phases of logic. Also, updated with new Azure SDK API.